ydb/core/mind/hive/hive_impl.cpp

```diff
@@ -3013,6 +3018,18 @@ void THive::UpdatePiles() {
     Execute(CreateUpdatePiles());
 }
 
+std::optional<TActorId> THive::GetPipeToTenantHive(const TNodeInfo* nodeInfo) {
+    if (nodeInfo && nodeInfo->ServicedDomains.size() == 1) {
+        TDomainInfo* domainInfo = FindDomain(nodeInfo->ServicedDomains.front());
+        if (domainInfo != nullptr) {
+            if (domainInfo->HiveId != 0 && domainInfo->HiveId != TabletID()) {
+                return domainInfo->GetPipeToHive(this);
+            }
+        }
+    }
+    return std::nullopt;
```

Comment:
Имхо, лучше инвертировать:
```suggestion
    if (!nodeInfo || nodeInfo->ServicedDomains.size() != 1) {
        return std::nullopt;
    }
    TDomainInfo* domainInfo = FindDomain(nodeInfo->ServicedDomains.front());
    if (!domainInfo || !domainInfo->HiveId || domainInfo->HiveId == TabletID()) {
        return std::nullopt;
    }

    return domainInfo->GetPipeToHive(this);
}
```